### PR TITLE
[TACHYON-959] Remove TachyonConf from parameter list of LocalTachyonCluster#start

### PR DESCRIPTION
**Pull request by uronce-cc**
https://tachyon.atlassian.net/browse/TACHYON-959

Fix several TODOs left.

### CODE REVIEW

File: minicluster/src/main/java/tachyon/master/LocalTachyonCluster.java [hunk 1]
```diff
@@ -234,33 +234,22 @@ public void run() {
   }
 
   /**
-   * Start both a master and a worker using the default configuration.
+   * Start both a master and a worker using the configurations in {@link MasterContext} and
```
Comment (yupeng9) [inline minicluster/src/main/java/tachyon/master/LocalTachyonCluster.java]: Start -> Starts

**Comment by yupeng9** [comment]
Thanks for the cleanup. I left one minor comment.

File: minicluster/src/main/java/tachyon/master/LocalTachyonCluster.java (hunk 1 above)
Comment (uronce-cc) [inline minicluster/src/main/java/tachyon/master/LocalTachyonCluster.java]: done.

**Comment by gpang** [comment]
Jenkins, test this please

**Comment by gpang** [comment]
Jenkins, test this please

**Comment by haoyuan** [comment]
Jenkins, test this please.

File: common/src/main/java/tachyon/worker/WorkerClient.java [hunk 2]
```diff
@@ -381,7 +381,7 @@ public synchronized String requestBlockLocation(long blockId, long initialBytes)
     try {
       return mClient.requestBlockLocation(mSessionId, blockId, initialBytes);
     } catch (OutOfSpaceException e) {
-      throw new IOException(e);
+      throw new IOException("Fail to request " + initialBytes, e);
```
Comment (jsimsa) [inline common/src/main/java/tachyon/worker/WorkerClient.java]: ``Failed to request``

File: minicluster/src/main/java/tachyon/master/LocalTachyonCluster.java [hunk 3]
```diff
@@ -234,33 +234,22 @@ public void run() {
   }
 
   /**
-   * Start both a master and a worker using the default configuration.
+   * Starts both a master and a worker using the configurations in {@link MasterContext} and
+   * {@link WorkerContext}.
```
Comment (jsimsa) [inline minicluster/src/main/java/tachyon/master/LocalTachyonCluster.java]: ``{@link MasterContext} and {@link WorkerContext} respectively.``

**Comment by jsimsa** [comment]
Cheng, thank you for the cleanup. Left you some minor comments. Also, please resolve the merge conflicts.

**Comment by uronce-cc** [comment]
@jsimsa thanks for review, all comments resolved, PTAL.

File: common/src/main/java/tachyon/worker/WorkerClient.java [hunk 4]
```diff
@@ -381,7 +381,7 @@ public synchronized String requestBlockLocation(long blockId, long initialBytes)
     try {
       return mClient.requestBlockLocation(mSessionId, blockId, initialBytes);
     } catch (OutOfSpaceException e) {
-      throw new IOException(e);
+      throw new IOException("Failed to request " + initialBytes, e);
```
Comment (gpang) [inline common/src/main/java/tachyon/worker/WorkerClient.java]: This change seems unrelated to this PR?

File: integration-tests/src/test/java/tachyon/client/IsolatedTachyonFileSystemIntegrationTest.java [hunk 5]
```diff
@@ -57,10 +58,13 @@ public final void after() throws Exception {
 
   @Before
   public final void before() throws Exception {
-    System.setProperty(Constants.USER_FILE_BUFFER_BYTES, Integer.toString(USER_QUOTA_UNIT_BYTES));
+    MasterContext.getConf().set(Constants.USER_FILE_BUFFER_BYTES, Integer.toString(
```
Comment (gpang) [inline integration-tests/src/test/java/tachyon/client/IsolatedTachyonFileSystemIntegrationTest.java]: Does changing the master conf affect the client conf? (I'm assuming USER_FILE_BUFFER_BYTES is some client parameter)

File: integration-tests/src/test/java/tachyon/client/RemoteBlockInStreamIntegrationTest.java [hunk 6]
```diff
@@ -98,20 +99,19 @@ public RemoteBlockInStreamIntegrationTest(String dataServer, String transferType
...
-    System.clearProperty(Constants.WORKER_NETWORK_NETTY_FILE_TRANSFER_TYPE);
-    System.clearProperty(Constants.USER_REMOTE_BLOCK_READER);
   }
 
   @Before
   public final void before() throws Exception {
+    TachyonConf conf = WorkerContext.getConf();
+    conf.set(Constants.WORKER_DATA_SERVER, mDataServerClass);
+    conf.set(Constants.WORKER_NETWORK_NETTY_FILE_TRANSFER_TYPE, mNettyTransferType);
+    conf.set(Constants.USER_REMOTE_BLOCK_READER, mRemoteReaderClass);
```
Comment (gpang) [inline integration-tests/src/test/java/tachyon/client/RemoteBlockInStreamIntegrationTest.java]: Will setting this for the worker context conf affect these user/client parameters? Can you verify this will work?

File: integration-tests/src/test/java/tachyon/client/TachyonFileSystemIntegrationTest.java [hunk 7]
```diff
@@ -69,7 +70,8 @@ public static final void afterClass() throws Exception {
 
   @BeforeClass
   public static void beforeClass() throws Exception {
-    System.setProperty(Constants.USER_FILE_BUFFER_BYTES, Integer.toString(USER_QUOTA_UNIT_BYTES));
+    MasterContext.getConf().set(Constants.USER_FILE_BUFFER_BYTES, Integer.toString(
```
Comment (gpang) [inline integration-tests/src/test/java/tachyon/client/TachyonFileSystemIntegrationTest.java]: Will modifying the master context conf affect this user parameter?

File: integration-tests/src/test/java/tachyon/worker/BlockServiceHandlerIntegrationTest.java [hunk 8]
```diff
@@ -78,11 +79,11 @@ public final void after() throws Exception {
 
   @Before
   public final void before() throws Exception {
-    TachyonConf tachyonConf = new TachyonConf();
+    TachyonConf tachyonConf = MasterContext.getConf();
     tachyonConf.set(Constants.USER_FILE_BUFFER_BYTES, String.valueOf(100));
```
Comment (gpang) [inline integration-tests/src/test/java/tachyon/worker/BlockServiceHandlerIntegrationTest.java]: Does changing this parameter in MasterContext affect the client?

File: integration-tests/src/test/java/tachyon/worker/DataServerIntegrationTest.java [hunk 9]
```diff
@@ -138,16 +135,16 @@ private void assertValid(final DataServerMessage msg, final int expectedSize, fi
 
   @Before
   public final void before() throws Exception {
-    TachyonConf tachyonConf = new TachyonConf();
+    TachyonConf tachyonConf = WorkerContext.getConf();
     tachyonConf.set(Constants.USER_FILE_BUFFER_BYTES, String.valueOf(100));
+    tachyonConf.set(Constants.WORKER_DATA_SERVER, mDataServerClass);
+    tachyonConf.set(Constants.WORKER_NETWORK_NETTY_FILE_TRANSFER_TYPE, mNettyTransferType);
+    tachyonConf.set(Constants.USER_REMOTE_BLOCK_READER, mBlockReader);
```
Comment (gpang) [inline integration-tests/src/test/java/tachyon/worker/DataServerIntegrationTest.java]: Does setting this in the WorkerContext affect the client?

File: integration-tests/src/test/java/tachyon/worker/block/meta/TieredStoreIntegrationTest.java [hunk 10]
```diff
@@ -65,12 +66,12 @@ public final void after() throws Exception {
 
   @Before
   public final void before() throws Exception {
-    TachyonConf tachyonConf = new TachyonConf();
+    TachyonConf tachyonConf = MasterContext.getConf();
     tachyonConf.set(Constants.USER_FILE_BUFFER_BYTES, String.valueOf(100));
```
Comment (gpang) [inline integration-tests/src/test/java/tachyon/worker/block/meta/TieredStoreIntegrationTest.java]: Does changing this user parameter in MasterContext work?

File: minicluster/src/main/java/tachyon/master/LocalTachyonClusterMultiMaster.java [hunk 11]
```diff
@@ -109,7 +109,6 @@ public synchronized TachyonFileSystem getClient() throws IOException {
     return mClientPool.getClient(mMasterConf);
   }
 
-  // TODO(cc): Since we have MasterContext now, remove this.
   public TachyonConf getMasterTachyonConf() {
```
Comment (gpang) [inline minicluster/src/main/java/tachyon/master/LocalTachyonClusterMultiMaster.java]: Can this method be removed?

**Comment by gpang** [comment]
I left several questions. Most of them were related to which context the parameters should be modified for. It is not obvious to me that changing worker or master contexts will affect the client. Thanks!

File: common/src/main/java/tachyon/worker/WorkerClient.java (hunk 4 above)
Comment (uronce-cc) [inline common/src/main/java/tachyon/worker/WorkerClient.java]: Jenkins fail previously, I thought this helps the debugging on the failed test.

File: minicluster/src/main/java/tachyon/master/LocalTachyonClusterMultiMaster.java (hunk 11 above)
Comment (uronce-cc) [inline minicluster/src/main/java/tachyon/master/LocalTachyonClusterMultiMaster.java]: Both this and the same one in `LocalTachyonCluster` can be removed, but I thought it's fine to leave them here since a lot of tests are using them...

**Comment by uronce-cc** [comment]
For client context, either set it in worker or master context will update the client context because of [this line](https://github.com/amplab/tachyon/blob/master/minicluster/src/main/java/tachyon/master/LocalTachyonCluster.java#L233).

**Comment by gpang** [comment]
LGTM
